app/logic/Contacts/JSON/JSONPerson.ts

```diff
@@ -45,6 +46,18 @@ export class JSONPerson {
     return json;
   }
 
+  static saveStreetAddresses(addresses: Collection<ContactEntry>): any[] {
+    return addresses.contents.map(c => this.saveAddress(c));
+  }
+
+  static saveAddress(contact: ContactEntry): any {
+    let json: any = new StreetAddress(contact.value).toJSON();
```

Comment:
Make this:
`json.value = new StreetAddress(contact.value).toJSON();`
please, so that the address is in its own JSON object within property `value`, not on the same level as `purpose`.

Reply:
Ugh, that's going to make those test files really ugly...